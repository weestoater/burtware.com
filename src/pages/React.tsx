import React from "react";
import ReactHooksPost from "../components/react/ReactHooksPost";
import RandomUsersPost from "../components/react/RandomUserPost";

export default function ReactPage() {
  return (
    <div className="row">
      <RandomUsersPost />
      <ReactHooksPost />
      <div className="col-lg-3 col-md-6 mb-4">
        <div className="card">
          <div className="card-header">
            <i className="bi bi-filetype-jsx" />
            <span className="pl-4">my experiences</span>
          </div>
          <div className="card-body">
            <p>
              There are as many ways to do Single Page Applications (SPAs) as
              there are developers with opinions. Within my work we have many
              instances of AngularJS, Angular (4,5,6,7,9,11,12) and ReactJS.
            </p>
            <p>
              My experience with ReactJS began as pretty limited, but with a
              strong following wind, a few cups of coffee and a Pluralsight
              subscription â€“ I resolved that in no small measure.</p>
            <p>
              Before moving recently to pure UX team, I worked on a growing React platform which included multiple chart and grid offerings.
            </p>
            <p>
              I am now trying to get my head around the worlds of <code>typescript</code>; micro-services and AWS with React.
            </p>
          </div>
        </div>
      </div>
    </div>
  );
}
