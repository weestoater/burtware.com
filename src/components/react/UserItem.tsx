import React from "react";

function UserItem(props: any) {

  const user = props;

  return (
    <>
      <div className="user col-sm-12 mb-3">
        <img src={user.item.picture.large} alt="" className="picture" />
        <p className="name">
          <span className="badge bg-info text-black me-2">{user.item.nat}</span>{" "}
          {user.item.name.title} {user.item.name.first} {user.item.name.last}
        </p>
        <address className="address">
          {user.item.location.street.number} {user.item.location.street.name}{" "}
          <br />
          {user.item.location.city}
          <br />
          {user.item.location.state}, {user.item.location.country}
          <br />
          {user.item.location.postcode}
        </address>
        <p className="contact">
          <i className="bi bi-phone" aria-hidden="true" />{" "}
          <a href={`tel:${user.item.cell}`}>{user.item.cell}</a>
          <br />
          <i className="bi bi-telephone" aria-hidden="true" />{" "}
          <a href={`tel:${user.item.phone}`}>{user.item.phone}</a>
          <br />
          <i className="bi bi-envelope" aria-hidden="true" />{" "}
          <a href={`mailto:${user.item.email}`}>{user.item.email}</a>
        </p>
      </div>
    </>
  );
}

export default UserItem;
