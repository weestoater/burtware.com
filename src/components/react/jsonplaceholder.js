import React, { useState, useEffect } from 'react';

export const JsonPlaceholder = () => {

    // const fetchData = fetch('https://geodb-cities-graphql.p.rapidapi.com', {
    //     method: 'POST',
    //     headers: {'Content-Type': 'application/json'},
    //     body: JSON.stringify({
    //         query:`{
    //             countries {
    //                 name
    //                 capital
    //                 currency
    //             }
    //         }`
    //     })
    // })
    // .then(res => res.json())
    // .then(res => console.log(res.data));

    // fetchData();

    return (
        <div className="card">
            <div className="card-header">FETCH DATA</div>
            <div className="card-body">
                body    
            </div> 
        </div>
    );
}