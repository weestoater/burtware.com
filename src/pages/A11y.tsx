import React from "react";
import A11yContent from "../components/a11y/A11yContent";

export default function A11y() {
  return (
    <div className="row">
      <A11yContent />
    </div>
  );
}
