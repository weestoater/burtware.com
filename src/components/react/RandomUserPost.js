import React, {useState, useEffect} from 'react';
import getApi from '../../adapters/commonApi';
import Spinner from '../global/Spinner';
import UserItem from './UserItem';

export default function RandomUsersPost() {

    const [users, setUsers] = useState({});
    const apiUrl = "https://randomuser.me/api/?results=3";

    useEffect(() => {
        getApi(apiUrl).then(setUsers);
      }, []);

    return (
        <div className="col-lg-4 col-md-6 mb-4">
            <div className="card">
                <div className="card-header">
                    <i className="bi bi-filetype-jsx" />
                    <span className="pl-4">Random Users API</span>
                </div>
                <div className="card-body users">
                {users.results ? (
                    users.results.map((user, key) => <UserItem key={key} item={user} /> )
                ) : ( 
                <Spinner /> 
                )}
                </div>
            </div>
        </div>
    );
}
