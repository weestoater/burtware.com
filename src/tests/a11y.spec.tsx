import React from "react";
import { render, screen } from "@testing-library/react";
import A11y from "../pages/A11y";

describe('Testing A11y page', () => {

    it('can render A11yPage', () => {
        render(<A11y />);
        const standards = screen.getByText(/standards/i);
        const accessibility = screen.getByText(/accessibility/i);
        expect(standards).toBeInTheDocument();
        expect(accessibility).toBeInTheDocument();
    });
});