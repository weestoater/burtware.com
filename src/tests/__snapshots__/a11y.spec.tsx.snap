// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing A11y page can render A11yPage 1`] = `
<div>
  <div
    class="row"
  >
    <div
      class="col-sm-12"
    >
      <h1
        class="visually-hidden"
      >
        Accessibility
      </h1>
    </div>
    <div
      class="col-lg-4 col-md-6"
    >
      <div
        class="card mb-4"
      >
        <div
          class="card-header"
        >
          <i
            class="bi bi-heart-pulse"
          />
           A11y & me ♿
        </div>
        <div
          class="card-body"
        >
          <p>
            I 
            <strong>
              really
            </strong>
             care about Accessibility (a11y) online and have done for over 20 years now.
          </p>
          <p>
            With a move to a new team in Sep '21, I've switched direction as a developer from being a delivery focused engineer, to working with teams as a SME 
            <small>
              (subject matter expert)
            </small>
             on UI development and more importantly Web Standards & A11Y.
          </p>
        </div>
      </div>
    </div>
    <div
      class="col-lg-4 col-md-6"
    >
      <div
        class="card mb-4"
      >
        <div
          class="card-header"
        >
          <i
            class="bi bi-award"
          />
           A11y & promotion ♿
        </div>
        <div
          class="card-body"
        >
          <p>
            I am incredibly lucky to work for a company who support and encourage my passion for A11y.  I was promoted to Vice President for the work I've been doing in this area and I am now even 
            <b>
              more
            </b>
             enthusiastic to prove how important it is and spread my net wider in the firm.
          </p>
          <p>
            I love engaging with teams and helping them discover the range of tools available and the steps they need to take to reach and keep A11y compliance. Often there are one or two developers who rise to the challenge and drive their team. This is the best result for everyone - as they can then carry the torch and keep a11y issues to a minimum.
          </p>
          <p>
            I was recently told by a fresh University graduate that A11y isn't even mentioned during a 
            <b>
              4 year
            </b>
             Computer Science degree - I am absolutely shocked to hear this as legislation has been in place for over 20+ years. With this in mind I am looking to become more involved in our Communities of Practise and POWERUP events internally to spread the good word and more importantly, offer education and advice to young or inexperienced developers.
          </p>
        </div>
      </div>
    </div>
    <div
      class="col-lg-4 col-md-6"
    >
      <div
        class="card mb-4"
      >
        <div
          class="card-header"
        >
          <i
            class="bi bi-tools"
          />
           Axe plugin ♿
        </div>
        <div
          class="card-body"
        >
          <p>
            I believe in clean easy to read code, especially in my 
            <code>
              html
            </code>
            . As part of my day job I help teams building web applications to make sure they are accessible and I always start off by getting their developers to install the 
            <a
              href="https://www.deque.com/axe/devtools/"
            >
              Axe Tools
            </a>
             for Chrome.
          </p>
          <p>
             This simple tool can give a great insight into issues teams are facing on the journey to a11y compliance.  I especially like the fact that there are the four bandings:
          </p>
          <ol>
            <li>
              Critial
            </li>
            <li>
              Serious
            </li>
            <li>
              Moderage
            </li>
            <li>
              Minor
            </li>
          </ol>
          <p>
            The ability to drill down on individual items and 
            <b>
              highlight
            </b>
             any offending code is brilliant and gets across the issues a lot better than any text book ever could.
          </p>
        </div>
      </div>
    </div>
  </div>
</div>
`;
